engine/gamesys/src/gamesys/test/collection_factory/collectionfactory_test.script

```diff
@@ -13,35 +13,82 @@
 -- specific language governing permissions and limitations under the License.
 
 
+local function state(self)
+    return self.states[self.state]
+end
+
+local function next_state(self)
+    self.state = self.state + 1
+end
+
+local function delete_gameobjects(self)
+    for _, id in ipairs(self.ids) do
+        go.delete(id, true)
+    end
+    self.ids = {}
+end
+
+local function handle_state(self)
+    local state = self.states[self.state]
+    -- print("Handle State", state)
+
+    if state == "load" then
```

Comment:
Reworked the test a bit to hanve more distinctly named states (as opposed to "step 1" etc)